Automatically sync TLSA records with Firefox cert_override.txt

JeremyRand opened the pull request.

hlandau commented on server/server.go:
```diff
@@ -213,6 +214,10 @@ func (s *Server) Start() error {
 	s.wgStart.Wait()
 	log.Info("Listeners started")
+
+	err := tlsoverridefirefoxsync.OverrideSyncStart(s.cfg.CanonicalSuffix)
```
style: avoid stutter: s/OverrideSyncStart/Start/

JeremyRand replied:
Fixed.

hlandau commented on server/server.go:
```diff
@@ -213,6 +214,10 @@ func (s *Server) Start() error {
+
+	err := tlsoverridefirefoxsync.OverrideSyncStart(s.cfg.CanonicalSuffix)
+	log.Fatale(err, "Couldn't start Firefox override sync")
+
```
change to return err

JeremyRand replied:
Does the `service` package automatically exit when `Start()` returns a non-nil error?

JeremyRand replied:
Empirically the answer is yes.  I'll make the change.

JeremyRand replied:
Fixed.

hlandau commented on tlsoverridefirefox/filter.go:
```diff
@@ -0,0 +1,33 @@
+		hostAndPort := tabSplit[0]
+
+		colonSplit := strings.Split(hostAndPort, ":")
+		host := colonSplit[0]
```
change to use net.SplitHostPort (handles IPv6 addresses properly)

JeremyRand replied:
Good catch.  I wasn't expecting IPv6 syntax to matter here since public CA's don't issue TLS certs for IP addresses, but now I realize that of course it's entirely plausible for a Firefox user to configure a certificate override for an IP address even though the cert isn't issued to the IP.

JeremyRand replied:
Fixed.

hlandau commented on tlsoverridefirefox/tlsoverridefirefoxsync/firefoxoverridesync.go:
```diff
@@ -0,0 +1,132 @@
+		"cert_override.txt")
+var firefoxProfileDirFlag = cflag.String(flagGroup, "profiledir", "",
+	"Firefox profile directory")
+
```
style: use var ( ... ) group

JeremyRand replied:
Fixed.

hlandau commented on tlsoverridefirefox/tlsoverridefirefoxsync/firefoxoverridesync.go:
```diff
@@ -0,0 +1,132 @@
+		cmdDumpZone := exec.Command("ncdumpzone",
+			"--ncdumpzone.format", "firefox-override")
+		zoneOverrides, err := cmdDumpZone.Output()
+		log.Fatale(err, "Couldn't dump zone for Firefox override sync")
```
don't do this; refactor ncdumpzone as a library and call it from here and the command line tool

JeremyRand replied:
Fixed, pending merge of https://github.com/namecoin/ncdns/pull/82 .

JeremyRand commented:
Thanks for the review @hlandau , I'll fix the issues you identified soon.

JeremyRand commented:
Okay, I think I've addressed all the feedback.  This PR is now ready for review again, but shouldn't be merged until https://github.com/namecoin/ncdns/pull/82 is merged.

JeremyRand commented:
#82 is merged.  @hlandau this PR is ready for review.  Once you've reviewed my fixes for your feedback, I'll squash.

hlandau commented:
ACK https://github.com/namecoin/ncdns/pull/80/commits/95a53599c73c59b9492c176ebd96916001ee375c

JeremyRand commented:
Squashed.

JeremyRand commented:
Added a documentation fix.

hlandau commented:
ACK https://github.com/namecoin/ncdns/pull/80/commits/2fc7fa5de6951f25114e041bdd667d4bc7e04df8

JeremyRand commented:
Squashed one last time.
